APIKit/APIKit.swift

```diff
@@ -119,4 +192,27 @@ public class API {
             return nil
         }
     }
+    
+    // MARK: - NSURLSessionDelegate
+    public func URLSession(session: NSURLSession, didReceiveChallenge challenge: NSURLAuthenticationChallenge, completionHandler: (NSURLSessionAuthChallengeDisposition, NSURLCredential!) -> Void) {
+        completionHandler(.PerformDefaultHandling, nil)
+    }
+    
+    // MARK: - NSURLSessionTaskDelegate
+    // TODO: add attributes like NS_REQUIRES_SUPER when it is available in future version of Swift.
+    public func URLSession(session: NSURLSession, task: NSURLSessionTask, didCompleteWithError connectionError: NSError?) {
+        if let dataTask = task as? NSURLSessionDataTask {
+            dataTask.completionHandler?(dataTask.responseBuffer, dataTask.response, connectionError)
+        }
+    }
+
+    // MARK: - NSURLSessionDataDelegate
+    public func URLSession(session: NSURLSession, dataTask: NSURLSessionDataTask, didReceiveResponse response: NSURLResponse, completionHandler: (NSURLSessionResponseDisposition) -> Void) {
```

Comment:
This is an optional delegate method. Should we need to set up default implementation? The document for this says:

> If you do not provide this delegate method, the session always allows the task to continue.